lib/url.js

```diff
@@ -539,13 +536,11 @@ function autoEscapeStr(rest) {
         break;
     }
   }
-  if (lastEscapedPos === 0)  // Nothing has been escaped.
-    return;
```

Comment:
You might want to check if it is a good idea to keep the if but return rest here.

Reply:
Yes, I tried with this check and here is the perf result:

```
cat result.csv | Rscript benchmark/compare.R
                                                                              improvement confidence   p.value
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="auth"           -2.80 %            0.8940666
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="dot"            10.52 %            0.5992881
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="file"           -5.34 %            0.7988524
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="idn"            -3.03 %            0.8843592
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="javascript"      7.21 %            0.7159490
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="long"           -3.54 %            0.8643568
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="percent"        11.36 %            0.5700568
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="short"          -2.29 %            0.9104903
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="ws"             11.91 %            0.5628974
```

Reply:
I retry with --runs 100 and now this is better:

```
cat result.csv | Rscript benchmark/compare.R
                                                                              improvement confidence    p.value
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="auth"           -1.44 %            0.11534475
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="dot"            -0.21 %            0.82526706
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="file"           -0.60 %            0.51438710
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="idn"            -1.96 %            0.05150804
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="javascript"     -1.01 %            0.30975298
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="long"           -0.10 %            0.90420107
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="percent"        -0.25 %            0.79938735
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="short"          -0.98 %            0.33971351
 url/legacy-vs-whatwg-url-parse.js n=100000 method="legacy" type="ws"             -0.97 %            0.26316218
```